# Olympic Medals Analysis

As performed in the coursera course assignment undertaken.

The data about to be imported from olympics.csv was taken from the coursera course which was derived from a wikipedia entry on All Time Olympic Games Medals. Some data cleaning will be done for some practice.

In [2]:
import pandas as pd

In [12]:
df = pd.read_csv('olympics.csv', index_col=0, skiprows=1)

In [13]:
df.head(8)

№ Summer  01 !  02 !  03 !  Total  № Winter  \
Afghanistan (AFG)                13     0     0     2      2         0   
Algeria (ALG)                    12     5     2     8     15         3   
Argentina (ARG)                  23    18    24    28     70        18   
Armenia (ARM)                     5     1     2     9     12         6   
Australasia (ANZ) [ANZ]           2     3     4     5     12         0   
Australia (AUS) [AUS] [Z]        25   139   152   177    468        18   
Austria (AUT)                    26    18    33    35     86        22   
Azerbaijan (AZE)                  5     6     5    15     26         5   

                           01 !.1  02 !.1  03 !.1  Total.1  № Games  01 !.2  \
Afghanistan (AFG)               0       0       0        0       13       0   
Algeria (ALG)                   0       0       0        0       15       5   
Argentina (ARG)                 0       0       0        0       41      18   
Armenia (ARM)                   0       0       0        0       11       1   
Australasia (ANZ) [ANZ]         0       0       0        0        2       3   
Australia (AUS) [AUS] [Z]       5       3       4       12       43     144   
Austria (AUT)                  59      78      81      218       48      77   
Azerbaijan (AZE)                0       0       0        0       10       6   

                           02 !.2  03 !.2  Combined total  
Afghanistan (AFG)               0       2               2  
Algeria (ALG)                   2       8              15  
Argentina (ARG)                24      28              70  
Armenia (ARM)                   2       9              12  
Australasia (ANZ) [ANZ]         4       5              12  
Australia (AUS) [AUS] [Z]     155     181             480  
Austria (AUT)                 111     116             304  
Azerbaijan (AZE)                5      15              26

In [14]:
for col in df.columns:
    if col[:2] == '01':
        df.rename(columns={col:'Gold'+col[4:]}, inplace=True)
    if col[:2] =='02':
        df.rename(columns={col:'Silver'+col[4:]}, inplace=True)
    if col[:2] =='03':
        df.rename(columns={col:'Bronze'+col[4:]}, inplace=True)
    if col[:1]=='№':
        df.rename(columns={col:'#'+col[1:]}, inplace=True)


In [16]:
df.head(8)

# Summer  Gold  Silver  Bronze  Total  # Winter  \
Afghanistan (AFG)                13     0       0       2      2         0   
Algeria (ALG)                    12     5       2       8     15         3   
Argentina (ARG)                  23    18      24      28     70        18   
Armenia (ARM)                     5     1       2       9     12         6   
Australasia (ANZ) [ANZ]           2     3       4       5     12         0   
Australia (AUS) [AUS] [Z]        25   139     152     177    468        18   
Austria (AUT)                    26    18      33      35     86        22   
Azerbaijan (AZE)                  5     6       5      15     26         5   

                           Gold.1  Silver.1  Bronze.1  Total.1  # Games  \
Afghanistan (AFG)               0         0         0        0       13   
Algeria (ALG)                   0         0         0        0       15   
Argentina (ARG)                 0         0         0        0       41   
Armenia (ARM)                   0         0         0        0       11   
Australasia (ANZ) [ANZ]         0         0         0        0        2   
Australia (AUS) [AUS] [Z]       5         3         4       12       43   
Austria (AUT)                  59        78        81      218       48   
Azerbaijan (AZE)                0         0         0        0       10   

                           Gold.2  Silver.2  Bronze.2  Combined total  
Afghanistan (AFG)               0         0         2               2  
Algeria (ALG)                   5         2         8              15  
Argentina (ARG)                18        24        28              70  
Armenia (ARM)                   1         2         9              12  
Australasia (ANZ) [ANZ]         3         4         5              12  
Australia (AUS) [AUS] [Z]     144       155       181             480  
Austria (AUT)                  77       111       116             304  
Azerbaijan (AZE)                6         5        15              26

In [18]:
names_ids = df.index.str.split('\s\(')
names_ids

Index([                           ['Afghanistan', 'AFG)'],
                                      ['Algeria', 'ALG)'],
                                    ['Argentina', 'ARG)'],
                                      ['Armenia', 'ARM)'],
                            ['Australasia', 'ANZ) [ANZ]'],
                          ['Australia', 'AUS) [AUS] [Z]'],
                                      ['Austria', 'AUT)'],
                                   ['Azerbaijan', 'AZE)'],
                                      ['Bahamas', 'BAH)'],
                                      ['Bahrain', 'BRN)'],
       ...
                                   ['Uzbekistan', 'UZB)'],
                                    ['Venezuela', 'VEN)'],
                                      ['Vietnam', 'VIE)'],
                               ['Virgin Islands', 'ISV)'],
                             ['Yugoslavia', 'YUG) [YUG]'],
       ['Independent Olympic Participants', 'IOP) [IOP]'],
                                 ['Zambia', '

In [19]:
df.index = names_ids.str[0]

In [20]:
df.head(8)

# Summer  Gold  Silver  Bronze  Total  # Winter  Gold.1  \
Afghanistan        13     0       0       2      2         0       0   
Algeria            12     5       2       8     15         3       0   
Argentina          23    18      24      28     70        18       0   
Armenia             5     1       2       9     12         6       0   
Australasia         2     3       4       5     12         0       0   
Australia          25   139     152     177    468        18       5   
Austria            26    18      33      35     86        22      59   
Azerbaijan          5     6       5      15     26         5       0   

             Silver.1  Bronze.1  Total.1  # Games  Gold.2  Silver.2  Bronze.2  \
Afghanistan         0         0        0       13       0         0         2   
Algeria             0         0        0       15       5         2         8   
Argentina           0         0        0       41      18        24        28   
Armenia             0         0        0       11       1         2         9   
Australasia         0         0        0        2       3         4         5   
Australia           3         4       12       43     144       155       181   
Austria            78        81      218       48      77       111       116   
Azerbaijan          0         0        0       10       6         5        15   

             Combined total  
Afghanistan               2  
Algeria                  15  
Argentina                70  
Armenia                  12  
Australasia              12  
Australia               480  
Austria                 304  
Azerbaijan               26

The columns are organised as # of Summer Games, Summer medals, # of Winter Games, Winter Medals, total # number of games, total # of medals.

In [21]:
df['ID'] = names_ids.str[1].str[:3]

In [22]:
df.head(8)

# Summer  Gold  Silver  Bronze  Total  # Winter  Gold.1  \
Afghanistan        13     0       0       2      2         0       0   
Algeria            12     5       2       8     15         3       0   
Argentina          23    18      24      28     70        18       0   
Armenia             5     1       2       9     12         6       0   
Australasia         2     3       4       5     12         0       0   
Australia          25   139     152     177    468        18       5   
Austria            26    18      33      35     86        22      59   
Azerbaijan          5     6       5      15     26         5       0   

             Silver.1  Bronze.1  Total.1  # Games  Gold.2  Silver.2  Bronze.2  \
Afghanistan         0         0        0       13       0         0         2   
Algeria             0         0        0       15       5         2         8   
Argentina           0         0        0       41      18        24        28   
Armenia             0         0        0       11       1         2         9   
Australasia         0         0        0        2       3         4         5   
Australia           3         4       12       43     144       155       181   
Austria            78        81      218       48      77       111       116   
Azerbaijan          0         0        0       10       6         5        15   

             Combined total   ID  
Afghanistan               2  AFG  
Algeria                  15  ALG  
Argentina                70  ARG  
Armenia                  12  ARM  
Australasia              12  ANZ  
Australia               480  AUS  
Austria                 304  AUT  
Azerbaijan               26  AZE

In [45]:
df.drop('Totals', inplace=True)

### Test: Get first country in data

In [35]:
df.iloc[0]

# Summer           13
Gold                0
Silver              0
Bronze              2
Total               2
# Winter            0
Gold.1              0
Silver.1            0
Bronze.1            0
Total.1             0
# Games            13
Gold.2              0
Silver.2            0
Bronze.2            2
Combined total      2
ID                AFG
Name: Afghanistan, dtype: object

In [37]:
df.index[0]

'Afghanistan'

### Which country has won the most gold medals in summer games?

In [49]:
df[df['Gold'] == df['Gold'].max()]

# Summer  Gold  Silver  Bronze  Total  # Winter  Gold.1  \
United States        26   976     757     666   2399        22      96   

               Silver.1  Bronze.1  Total.1  # Games  Gold.2  Silver.2  \
United States       102        84      282       48    1072       859   

               Bronze.2  Combined total   ID  
United States       750            2681  USA

In [50]:
df[df['Gold'] == df['Gold'].max()].index[0]

'United States'

Quick way of getting index of something with the max or min value.

In [70]:
df['Gold'].idxmax()

'United States'

Out of curiosity which countries have the most summer gold medals?

In [98]:
df_summer_gold = df.copy()

In [99]:
df_summer_gold.sort_values('Gold', ascending=False)['Gold'].head(10)

United States    976
Soviet Union     395
Great Britain    236
France           202
China            201
Italy            198
Germany          174
Hungary          167
East Germany     153
Sweden           143
Name: Gold, dtype: int64

### Which country has the biggest difference between their summer and winter gold medal counts?

In [51]:
df['gold_diff'] = abs(df['Gold'] - df['Gold.1'])

In [69]:
df['gold_diff'].idxmax()

'United States'

In [54]:
df[df['gold_diff'] == df['gold_diff'].max()]

# Summer  Gold  Silver  Bronze  Total  # Winter  Gold.1  \
United States        26   976     757     666   2399        22      96   

               Silver.1  Bronze.1  Total.1  # Games  Gold.2  Silver.2  \
United States       102        84      282       48    1072       859   

               Bronze.2  Combined total   ID  gold_diff  
United States       750            2681  USA        880

The United States has the largest absolute difference between number of Summer and Winter Gold Medals.

Out of curiosity, which countries do have the largest gap?

In [74]:
df_temp = df.copy()
df_temp.sort_values('gold_diff', ascending=False).head(10)

# Summer  Gold  Silver  Bronze  Total  # Winter  Gold.1  \
United States        26   976     757     666   2399        22      96   
Soviet Union          9   395     319     296   1010         9      78   
Great Britain        27   236     272     272    780        22      10   
China                 9   201     146     126    473        10      12   
France               27   202     223     246    671        22      31   
Hungary              25   167     144     165    476        22       0   
Italy                26   198     166     185    549        22      37   
Australia            25   139     152     177    468        18       5   
Japan                21   130     126     142    398        20      10   
East Germany          5   153     129     127    409         6      39   

               Silver.1  Bronze.1  Total.1  # Games  Gold.2  Silver.2  \
United States       102        84      282       48    1072       859   
Soviet Union         57        59      194       18     473       376   
Great Britain         4        12       26       49     246       276   
China                22        19       53       19     213       168   
France               31        47      109       49     233       254   
Hungary               2         4        6       47     167       146   
Italy                34        43      114       48     235       200   
Australia             3         4       12       43     144       155   
Japan                17        18       45       41     140       143   
East Germany         36        35      110       11     192       165   

               Bronze.2  Combined total   ID  gold_diff  
United States       750            2681  USA        880  
Soviet Union        355            1204  URS        317  
Great Britain       284             806  GBR        226  
China               145             526  CHN        189  
France              293             780  FRA        171  
Hungary             169             482  HUN        167  
Italy               228             663  ITA        161  
Australia           181             480  AUS        134  
Japan               160             443  JPN        120  
East Germany        162             519  GDR        114

### Which country has the biggest difference between their summer gold medal counts and winter gold medal counts relative to total gold medal count?

In [75]:
df_won_both = df_temp[(df_temp['Gold'] > 0) & (df_temp['Gold.1'] > 0)]

In [78]:
df_won_both['relative_difference_gold'] = df_won_both.gold_diff/df_won_both['Gold.2']

/Users/aarond/dev/envs/000-quick-scripts/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [80]:
df_won_both['relative_difference_gold'].idxmax()

'Bulgaria'

In [85]:
df_won_both[df_won_both['relative_difference_gold'] == df_won_both['relative_difference_gold'].max()][['Gold', 'Gold.1', 'Gold.2', 'gold_diff', 'relative_difference_gold']]

Gold  Gold.1  Gold.2  gold_diff  relative_difference_gold
Bulgaria    51       1      52         50                  0.961538

Bulgaria has the largest difference between summer gold and winter gold medal counts, relative to the total gold medal count, by winning 51 golds in Summer Olympics while only winning 1 gold in Winter Olympics (including only counties who have won at least one of both).

Out of curiosity what are the top countries with the largest relative difference in gold medals.

In [84]:
df_won_both.sort_values('relative_difference_gold', ascending=False).head(10)[['Gold', 'Gold.1', 'Gold.2', 'gold_diff', 'relative_difference_gold']]

Gold  Gold.1  Gold.2  gold_diff  relative_difference_gold
Bulgaria          51       1      52         50                  0.961538
Belgium           37       1      38         36                  0.947368
Spain             37       1      38         36                  0.947368
Australia        139       5     144        134                  0.930556
Czechoslovakia    49       2      51         47                  0.921569
Great Britain    236      10     246        226                  0.918699
China            201      12     213        189                  0.887324
Ukraine           33       2      35         31                  0.885714
Kazakhstan        16       1      17         15                  0.882353
Japan            130      10     140        120                  0.857143

### Tally the points of all country medals by assigning 3 points for gold, 2 for silver, and 1 for bronze.

In [86]:
df['points'] = df['Gold.2']*3 + df['Silver.2']*2 + df['Bronze.2']

In [90]:
df['points'].sort_values(ascending=False)[:10]

United States    5684
Soviet Union     2526
Great Britain    1574
Germany          1546
France           1500
Italy            1333
Sweden           1217
China            1120
East Germany     1068
Russia           1042
Name: points, dtype: int64

Tally the points of all medals, assigning the points of 3 per gold, 2 per silver, and 1 per bronze, the United States has the most points of any country in the word with 5684 points. This is more than double of the second highest being the Soviet Union, and more than triple of the third higest (Great Britain). 

Other curious questions.

### Countries with the most winter olympic gold?

In [100]:
df_winter_gold = df.copy()
df_winter_gold.sort_values('Gold.1', ascending=False)['Gold.1'].head(10)

Norway           118
United States     96
Soviet Union      78
Germany           78
Canada            62
Austria           59
Switzerland       50
Sweden            50
Russia            49
Finland           42
Name: Gold.1, dtype: int64

Norway has the highest number of winter olympic gold at 118 gold medals. 

The United States have the second highest at 96, which is impressive considering that that also have the largest absolute gap for summer and gold medals among all countries. However the impressive results of the US like other nations such as the UK are also helped due to the number of olympics they have participated in, giving them a higher chance of obtaining more gold or any medals for that matter than any other country/nation. 

### Which countries have participated in the most olympics?

In [101]:
df_number_of_olympics = df.copy()

In [104]:
df_number_of_olympics.sort_values('# Games', ascending=False).head(10)

# Summer  Gold  Silver  Bronze  Total  # Winter  Gold.1  \
Great Britain        27   236     272     272    780        22      10   
Switzerland          27    47      73      65    185        22      50   
France               27   202     223     246    671        22      31   
Austria              26    18      33      35     86        22      59   
Sweden               26   143     164     176    483        22      50   
United States        26   976     757     666   2399        22      96   
Italy                26   198     166     185    549        22      37   
Hungary              25   167     144     165    476        22       0   
Canada               25    59      99     121    279        22      62   
Norway               24    56      49      43    148        22     118   

               Silver.1  Bronze.1  Total.1  # Games  Gold.2  Silver.2  \
Great Britain         4        12       26       49     246       276   
Switzerland          40        48      138       49      97       113   
France               31        47      109       49     233       254   
Austria              78        81      218       48      77       111   
Sweden               40        54      144       48     193       204   
United States       102        84      282       48    1072       859   
Italy                34        43      114       48     235       200   
Hungary               2         4        6       47     167       146   
Canada               56        52      170       47     121       155   
Norway              111       100      329       46     174       160   

               Bronze.2  Combined total   ID  gold_diff  points  
Great Britain       284             806  GBR        226    1574  
Switzerland         113             323  SUI          3     630  
France              293             780  FRA        171    1500  
Austria             116             304  AUT         41     569  
Sweden              230             627  SWE         93    1217  
United States       750            2681  USA        880    5684  
Italy               228             663  ITA        161    1333  
Hungary             169             482  HUN        167     962  
Canada              173             449  CAN          3     846  
Norway              143             477  NOR         62     985